### PR TITLE
bedrock: migrate usage extraction to `RequestUsage.extract()`

### PR DESCRIPTION
**Pull request by adtyavrdhn**
- Part of #4818 (bedrock item; the issue stays open for the remaining models)

Migrates Bedrock Converse usage mapping to `RequestUsage.extract()` via genai-prices, replacing the manual field mapping in both the sync and streaming paths (`count_tokens` is a different API shape and intentionally untouched).

Semantics are unchanged: AWS documents Converse `inputTokens` as excluding cache tokens, and the genai-prices aws extractor (0.0.70, pydantic/genai-prices#454) performs the same cache-inclusive add-back the manual code did, so all existing tests pass with zero snapshot changes. New mocked tests cover cache read/write extraction in both paths, including skipping the non-integer `cacheDetails` breakdown.

Requires genai-prices 0.0.70 (on 0.0.69 the aws default extractor dropped cache tokens entirely); main already pins `>=0.0.70` since #6330, so this is a pure migration with no dependency changes.

### Checklist

- [ ] Any **AI generated code** has been reviewed line-by-line by the human PR author, who stands by it.
- [x] No **breaking changes** in accordance with the [version policy](https://github.com/pydantic/pydantic-ai/blob/main/docs/version-policy.md).
- [x] **PR title** is fit for the [release changelog](https://github.com/pydantic/pydantic-ai/releases).

<a href="https://cubic.dev/pr/pydantic/pydantic-ai/pull/6328?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

### CODE REVIEW

File: tests/models/test_bedrock.py [hunk 1]
```diff
@@ -837,6 +837,95 @@ async def test_bedrock_unified_service_tier_auto_omits(
...
+    assert result.usage == snapshot(
+        RunUsage(
+            input_tokens=1524,
+            cache_write_tokens=7,
+            cache_read_tokens=1504,
+            output_tokens=5,
+            requests=1,
+            details={'futureBillableTokens': 11},
+        )
+    )
```
[the remark this answers is not in the document]
Comment (adtyavrdhn) [reply tests/models/test_bedrock.py:876]: Fixed in 7c45095d. Both mocked tests now explain why synthetic metadata requires mocks.
